docs/Makefile

```diff
@@ -34,3 +34,5 @@ plantuml_diags: $(SVG_FILES)
 
 clean:
 	rm -rf build/
+	# Delete files/dirs generated by the autosummary directive
+	find source -name generate -type d | xargs rm -rf
```

Comment:
Running find in makefiles can give lots of interesting performance and permission problems, can this be replaced with `git clean -dXf source` instead?

If it can't, I think `find source -name generate -type d -exec rm -rf {} \;` will at least give more descriptive error messages than piping everything to xargs.